core/src/main/java/com/amplifyframework/auth/result/AuthSignUpResult.java

```diff
@@ -19,7 +19,7 @@
 
 import com.amplifyframework.auth.AuthCodeDeliveryDetails;
 
-public final class AuthSignUpResult {
+public class AuthSignUpResult {
```

Comment:
Nothing in this PR actually extends it, though, right? Why open it?

Reply:
By design, we want to enable plugin developers to put any plugin specific return values in an extension of this class.

Reply:
My understanding is that, in general, Objects in the core library for a category should be non-final because, by nature, they are simply capturing what must be common across all plugins but we also expect there to be plugin specific values which we don't want be put up at the core level but rather extended at the plugin implementation level.

Reply:
[Open-closed principle](https://en.wikipedia.org/wiki/Open%E2%80%93closed_principle (of [SOLID](https://en.wikipedia.org/wiki/SOLID)) is:
> software entities (classes, modules, functions, etc.) should be open for extension, but closed for modification

In practice, there are two principle ways to achieve extension points: inheritance, and composition. Software texts like Design Patterns tell us to [prefer composition over inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance).

Even so, there are [benefits](https://en.wikipedia.org/wiki/Composition_over_inheritance#Benefits) and [drawbacks](https://en.wikipedia.org/wiki/Composition_over_inheritance#Drawbacks) to using composition over inheritance, and both can be valuable tools.

_Personally_, the issues I have with inheritance are mostly a few:
1. Base classes grow over time, to include muddled concerns; `AbstractFooBase` (what is it?)
2. With a closed class implementation, you can encapsulate all of the policy and assumptions you hold about the function of the software. When you open a class, there is no guarantee that the derived class will meet all of these assumptions.

An interface has problem (2). An open non-abstract class is susceptible to (1). An abstract class may be used to enforce policy while offering means for extension/composition, and so contains partial mitigation to both (1) and (2).

Ref:
- [Delegation pattern](https://en.wikipedia.org/wiki/Delegation_pattern)